### ISSUE TITLE
Cluster url auto changed from https to http

### ISSUE THREAD

**Issue by Shubham9t9** [kubecost/cost-analyzer#15]
Whenever I'm trying to add a context using https url, it gets changed to http and also marked as `unavailable context`

**Comment by dwbrown2** [comment]
@Shubham9t9 any more context you share on the issue? For example, can you share version + how you're adding additional contexts?

cc @Adam-Stack-PM to track

**Comment by Adam-Stack-PM** [comment]
@wolfeaustin, Chatted about this item this morning but could you triage this as part of testing the context switcher and update this issue based on your findings?

**Comment by Adam-Stack-PM** [comment]
@nealormsbee re routing to you for v1.98.  Based on Austin and my quick research, we may be hard coding http in a few places.

**Comment by nealormsbee** [comment]
Yep, that appears to be what's happening. I think I know what to do here -- making the change includes some possible risk but risk that I believe can be mitigated. Will write back here with an analysis and possibly a PR :+1:

**Comment by nealormsbee** [comment]
Rerouting this to @wolfeaustin. Here's the hardcoding I'm aware of:

In `src/pages/ClusterList/index.tsx` we have

```
function addNewCluster(newClusterAddress: string) {
    const address =
      'http://' +
      sanitizeHTML(newClusterAddress.trim())
        .replace(/^https?:\/\//, '')
        .replace(/\/$/, '');

    if (localClusterEndpoints.includes(address)) {
      return alert(`This address (${address}) has already been added`);
    }

    addContext(address);
    setFetchState(FetchStates.LOADING);
    setDialogOpen(false);
  }
}
```

Which means that any time we add a new cluster, we're stripping the protocol and prefixing it with `http://`.
I think what we'll want to do instead is:

- Check whether `newClusterAddress` starts with `http://` or `https://`
- If it does, then leave that protocol prefix alone, but still run `sanitizeHTML` and `replace(/\/$/, '')`
- If `newClusterAddress` does *not* start with `http://` or `https://`, then add a prefix of `http://`

Basically, if they don't supply a protocol prefix, use `http://` as a default. Otherwise use the protocol supplied.

Let me know if I can help with anything!
